src/Symfony/Component/Console/Output/Output.php

```diff
@@ -121,21 +121,39 @@ public function isDebug()
     /**
      * {@inheritdoc}
      */
-    public function writeln($messages, $type = self::OUTPUT_NORMAL)
+    public function writeln($messages, $options = self::OUTPUT_NORMAL)
     {
-        $this->write($messages, true, $type);
+        $this->write($messages, true, $options);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function write($messages, $newline = false, $type = self::OUTPUT_NORMAL)
+    public function write($messages, $newline = false, $options = self::OUTPUT_NORMAL)
     {
-        if (self::VERBOSITY_QUIET === $this->verbosity) {
-            return;
+        $messages = (array) $messages;
+
+        $type = self::OUTPUT_NORMAL;
+        if ($options & self::OUTPUT_RAW) {
+            $type = self::OUTPUT_RAW;
+        } elseif ($options & self::OUTPUT_PLAIN) {
+            $type = self::OUTPUT_PLAIN;
         }
 
-        $messages = (array) $messages;
+        $verbosity = self::VERBOSITY_NORMAL;
+        if ($options & self::VERBOSITY_QUIET) {
+            $verbosity = self::VERBOSITY_QUIET;
+        } elseif ($options & self::VERBOSITY_VERBOSE) {
+            $verbosity = self::VERBOSITY_VERBOSE;
+        } elseif ($options & self::VERBOSITY_VERY_VERBOSE) {
+            $verbosity = self::VERBOSITY_VERY_VERBOSE;
+        } elseif ($options & self::VERBOSITY_DEBUG) {
+            $verbosity = self::VERBOSITY_DEBUG;
+        }
```

Comment:
these if/elseif statements can be simplified significantly:

```php
$typeCheck = self::OUTPUT_NORMAL | self::OUTPUT_RAW | self::OUTPUT_PLAIN;
$type = $typeCheck & $options ?: self::OUTPUT_NORMAL;
$verbosity = ~$typeCheck & $options ?: self::VERBOSITY_NORMAL;
```

or (less dynamic):
```php
$type = 7 & $options ?: self::OUTPUT_NORMAL;
$verbosity = -8 & $options ?: self::VERBOSITY_NORMAL;
```